src/vivarium_dcpn_hypertension_mgt/components/treatment_algo.py

```diff
@@ -41,3 +44,65 @@ def get_rx_fill(self, index) -> pd.Series:
     def get_appt_followup(self, index) -> pd.Series:
         return self.rand_appt_followup.get_draw(index) < self.df_ad_data.appt_ad.loc[index]
 
+
+class MeasuredSBP:
+    @property
+    def name(self):
+        return "high_systolic_blood_pressure"
+
+    configuration_defaults = {
+        'measurement': {
+            'error_sd': 6,
+        }
+    }
+
+    def __init__(self):
+        self.configuration_defaults = {
+            f'{self.name}_measurement': MeasuredSBP.configuration_defaults['measurement']
+        }
+
+    def setup(self, builder: Builder):
+        self.measurement_error = builder.configuration[f'{self.name}_measurement'].error_sd
+        self.randomness = builder.randomness.get_stream(f'{self.name}.measurement')
+        self.true_exposure = builder.value.get_value(f'{self.name}.exposure')
+
+        self.measurement_column = f'{self.name}_measurement'
+        columns_created = [self.measurement_column]
+
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 creates_columns=columns_created)
+
+        self.population_view = builder.population.get_view(columns_created)
+
+        builder.event.register_listener('time_step__prepare', self.on_time_step_prepare)
+
+    def on_initialize_simulants(self, pop_data):
+        self.population_view.update(
+            pd.Series(np.nan, name=self.measurement_column, index=pop_data.index)
+        )
+
+    def on_time_step_prepare(self, event: Event):
+        self.population_view.update(pd.Series(np.nan, index=event.index, name=self.measurement_column))
+
+    def __call__(self, idx_measure: pd.Index, idx_record_these: pd.Index, measure_type_average: bool = False):
+        draw = self.randomness.get_draw(idx_measure)
+        if self.measurement_error:
+            noise = scipy.stats.norm.ppf(draw, scale=self.measurement_error)
+        else:
+            noise = 0
+
+        true_exp = self.true_exposure(idx_measure)
+        detect_zero = true_exp[true_exp==0]
+        hypertension_measurement = self.true_exposure(idx_measure) + noise
```

Comment:
I would use your `true_exp` variable here instead of recalling the pipeline - we haven't implemented caching of pipelines yet so every call requires recalculation, which for a risk exposure like this involves drawing from distributions.